route/include/route/route_generator_worker.hpp

```diff
@@ -16,67 +16,81 @@
 
-#include <string>
-#include <vector>
-#include <queue>
-#include <iostream>
-#include <fstream>
-#include <cav_msgs/Route.h>
-#include <cav_msgs/RouteEvent.h>
-#include <cav_msgs/RouteState.h>
-#include <cav_srvs/GetAvailableRoutes.h>
-#include <cav_srvs/SetActiveRoute.h>
-#include <cav_srvs/AbortActiveRoute.h>
-#include <carma_wm/WMListener.h>
-#include <carma_wm/WorldModel.h>
-#include <carma_wm/Geometry.h>
-#include <geometry_msgs/PoseStamped.h>
+#include <rclcpp/rclcpp.hpp>
+
+#include <carma_planning_msgs/msg/route.hpp>
+#include <carma_planning_msgs/msg/route_state.hpp>
+#include <carma_planning_msgs/msg/route_event.hpp>
+#include <geometry_msgs/msg/pose_stamped.hpp>
+#include <geometry_msgs/msg/twist_stamped.hpp>
+#include <visualization_msgs/msg/marker.hpp>
+#include <visualization_msgs/msg/marker_array.hpp>
+#include <std_msgs/msg/string.hpp>
+#include <carma_planning_msgs/srv/get_available_routes.hpp>
+#include <carma_planning_msgs/srv/set_active_route.hpp>
+#include <carma_planning_msgs/srv/abort_active_route.hpp>
+
+#include <carma_wm_ros2/WMListener.hpp>
+#include <carma_wm_ros2/WorldModel.hpp>
+#include <carma_wm_ros2/Geometry.hpp>
+
 #include <lanelet2_core/geometry/Lanelet.h>
 #include <lanelet2_core/primitives/Lanelet.h>
-#include <wgs84_utils_ros1/wgs84_utils.h>
+#include <lanelet2_extension/projection/local_frame_projector.h>
+#include <lanelet2_extension/io/autoware_osm_parser.h>
+#include <lanelet2_core/utility/Utilities.h>
+
+#include <wgs84_utils/wgs84_utils.h>
 #include <boost/filesystem.hpp>
 #include <boost/optional.hpp>
-#include <visualization_msgs/MarkerArray.h>
-#include <geometry_msgs/TwistStamped.h>
-#include <geometry_msgs/PoseStamped.h>
+#include <math.h>
 #include <unordered_set>
-#include <lanelet2_extension/projection/local_frame_projector.h>
-#include <lanelet2_extension/io/autoware_osm_parser.h>
 #include <functional>
-#include <std_msgs/String.h>
 #include <tf2_ros/transform_listener.h>
 #include <tf2/LinearMath/Transform.h>
 #include <tf2_geometry_msgs/tf2_geometry_msgs.h>
 
-
-#include "route_state_worker.h"
+#include "route/route_state_worker.hpp"
 
 namespace route {
 
     class RouteGeneratorWorker
     {
 
     public:
+        /**
+         * \brief Constructor for RouteGeneratorWorker class taking in dependencies via dependency injection
+         */
+        RouteGeneratorWorker(tf2_ros::Buffer& tf2_buffer);
+
         /**
          * \brief reroutingChecker function to set the rerouting flag locally
          */
         std::function<bool()> reroutingChecker;
+
         /**
          * \brief setReroutingChecker function to set the rerouting flag
          */
-        void setReroutingChecker(std::function<bool()> inputFunction);
-        
-        /**
-         * \brief Constructor for RouteGeneratorWorker class taking in dependencies via dependency injection
-         */
-        RouteGeneratorWorker() = default;
+        void setReroutingChecker(const std::function<bool()>& inputFunction);
```

Comment:
This is actually one thing sonar cloud is wrong about std::function should generally not be passed by reference. I'll mark this as won't fix in the sonar scan after you revert